docs/source-2.0/guides/client-guidance/identity-auth/index.md

```diff
@@ -0,0 +1,416 @@
+# Identity and Authentication
+
+Operations supported by a Smithy service generally require a mechanism for
+authenticating the client's identity. This section describes how to model the
+workflow of identity retrieval and request authentication in Smithy clients.
+
+## Identity
+
+An Identity is an entity representing **who the caller is**. Abstractly, the
+caller's identity could be anonymous, a token, a public/private key, etc.
+
+Identity types in the client runtime should typically extend from a base
+interface that exposes a single value: the expiration time of the identity. The
+API should then branch per auth scheme. Sometimes an Identity type may be
+re-used across multiple schemes.
+
+```java
+public interface Identity {
+	Instant expiration();
+}
+
+// for smithy.api#bearerTokenAuth
+public interface BearerTokenIdentity extends Identity {
+	String token();
+}
+
+// for aws.auth#sigv4, aws.auth#sigv4a
+public interface SigV4CredentialIdentity extends Identity {
+	String accessKeyId();
+	String secretAccessKey();
+	String sessionToken();
+}
+```
+
+## IdentityResolver
+
+An IdentityResolver implements the retrieval of a single type of identity.
+
+The IdentityResolver interface should be modeled as follows:
+
+```java
+public interface IdentityResolver<TIdentity extends Identity> {
+    TIdentity resolve(Object properties);
```

Comment:
The identity resolver / signer property bags are not formally defined here so I've just left them with an `Object` placeholder. I think the typed property bag construct needs its own section in these guidance docs.